andela/Ah-backend-guardians
# #162685668 Users can follow and unfollow

**What does this PR do?**
This PR enables users to follow each other's profiles and optionally unfollow each other.

**Description of Task to be completed?**
The application should enable a user to be able to follow another registered user's profile and unfollow them at will. The task should also enable a user to view their followers and other users that they're following.

**Link to pivotal tracker story**
[#162685668](https://www.pivotaltracker.com/story/show/162685668)

**How should this be manually tested?**

To follow a user, hit this endpoint, running a `PUT` request https://ah-backend-guardians-sta-pr-22.herokuapp.com/api/profile/follow/ and pass this model in the body.

Please note that you need to have registered this user first as it will give you an error message.

```
    {
        "username": "test"
    }
```

The output you should get back should be as in this **screenshot**.

![image](https://user-images.githubusercontent.com/38134382/51492471-0585e500-1dc3-11e9-9e42-e4c00dc3cbda.png)


To unfollow that user, just the request again and the output should appear as in the screenshot below.

![image](https://user-images.githubusercontent.com/38134382/51492527-441b9f80-1dc3-11e9-8ad4-33e6d20915d5.png)


To view all your followers, hit the endpoint https://ah-backend-guardians-sta-pr-22.herokuapp.com/api/profile/followers/ , running a `GET`request and you should get output that looks as in the **screenshot**.

![image](https://user-images.githubusercontent.com/38134382/51492773-01a69280-1dc4-11e9-8ff2-bfad98a8a0ff.png)


To view all the people you're following hit the endpoint https://ah-backend-guardians-sta-pr-22.herokuapp.com/api/profile/following/ , running a `GET`request and you should get back output as in the **screenshot**.

![image](https://user-images.githubusercontent.com/38134382/51492796-1420cc00-1dc4-11e9-8539-b0c59269cb80.png)
